apps/examples/src/3-custom-config/CustomConfigExample.tsx

```diff
@@ -1,120 +1,22 @@
 import '@tldraw/tldraw/editor.css'
 import '@tldraw/tldraw/ui.css'
+import { CardTool } from './CardTool'
+import { CardUtil } from './CardUtil'
 
-// Let's make a custom shape called a Card.
-
-// Shape Type
-// ----------
-// The shape type defines the card's type (`card`) and its props.
-// Every shape needs an opacity prop (for now), but other than that
-// you can add whatever you want, so long as it's JSON serializable.
-type CardShape = TLBaseShape<
-	'card',
-	{
-		w: number
-		h: number
-		opacity: TLOpacityType
-	}
->
-
-// Shape Util
-// ----------
-// The CardUtil class is used by the app to answer questions about a
-// shape of the 'card' type. For example, what is the default props
-// for this shape? What should we render for it, or for its indicator?
-class CardUtil extends TLBoxUtil<CardShape> {
-	static override type = 'card' as const
-
-	// There are a LOT of other things we could add here, like these flags
-	override isAspectRatioLocked = (_shape: CardShape) => false
-	override canResize = (_shape: CardShape) => true
-	override canBind = (_shape: CardShape) => true
-
-	override defaultProps(): CardShape['props'] {
-		return {
-			opacity: '1',
-			w: 300,
-			h: 300,
-		}
-	}
-
-	// This is the component that will be rendered for the shape.
-	// Try changing the contents of the HTMLContainer to see what happens.
-	render(shape: CardShape) {
-		// You can access class methods from here
-		const bounds = this.bounds(shape)
-
-		return (
-			<HTMLContainer
-				id={shape.id}
-				style={{
-					border: '1px solid black',
-					display: 'flex',
-					alignItems: 'center',
-					justifyContent: 'center',
-					pointerEvents: 'all',
-				}}
-			>
-				{/* Anything you want can go here—it's a regular React component */}
-				{bounds.w.toFixed()}x{bounds.h.toFixed()}
-			</HTMLContainer>
-		)
-	}
-
-	// The indicator is used when hovering over a shape or when it's selected.
-	// This can only be SVG path data; generally you want the outline of the
-	// component you're rendering.
-	indicator(shape: CardShape) {
-		return <rect width={shape.props.w} height={shape.props.h} />
-	}
-}
-
-// Tool
-// ----
-// Because the card tool can be just a rectangle, we can extend the
-// TLBoxTool class. This gives us a lot of functionality for free.
-export class CardTool extends TLBoxTool {
-	static override id = 'card'
-	static override initial = 'idle'
-	override shapeType = 'card'
-}
-
-// Finally, collect the custom tools and shapes into a config object
-const customTldrawConfig = new TldrawEditorConfig({
-	tools: [CardTool],
-	shapes: {
-		card: {
-			util: CardUtil,
-		},
-	},
-})
-
-// ... and we can make our custom shape example!
-export default function Example() {
+export default function CustomConfigExample() {
 	return (
 		<div className="tldraw__editor">
 			<Tldraw
-				persistenceKey="custom-config"
-				config={customTldrawConfig}
 				autoFocus
+				tools={[CardTool]}
+				shapes={{ card: { util: CardUtil } }}
```

Comment:
I don't think we should encourage people to construct these objects at render time.

With this, every time this component re-renders it'll re-mount the app. If you added any extra state here or further up the tree, then you'll lose app-only state whenever that state changes.

We could have people wrap these in `useMemo`? but that seems over complicated & probably having people construct them outside of the component and just pass them in is what we want to encourage